### DISCUSSION TITLE
Load model from jar file

### DISCUSSION THREAD

**Discussion by csxiang18** [deepjavalibrary/djl#787] [Q&A] [answered]
Hello, I want to package my model in a jar file.  I put `dnn-model/model.pt` under the resources directory. This works fine when I test it locally. However, when another project use this class, it could not load the model correctly. As the model is in the jar file which the urlString give like this: jar:file:/xxx/my-lib.jar!/dnn-model/.  Is there any way I could fix this?

    
    public static void loadDefaultDnnModel() {
        try {
            ClassLoader classLoader = DnnCalculator.class.getClassLoader();
            String urlString= classLoader.getResource("dnn-model").toString();
            logger.info("Load model from: " + urlString);
            Translator<Float[], Float[]> translator = new FloatToFloatTranslator();
            Criteria<Float[], Float[]> criteria = Criteria.builder()
                    .setTypes(Float[].class, Float[].class)
                    .optTranslator(translator)
                    .optModelUrls(urlString)
                    .optModelName("model")
                    .optProgress(new ProgressBar())
                    .build();
            ZooModel<Float[], Float[]> model = ModelZoo.loadModel(criteria);
            Predictor<Float[], Float[] > predictor = model.newPredictor();
            predictor.predict(new Float[]{1.0f, 2.0f});
        } catch (Exception e) {
            logger.info("Load default model error");
            e.printStackTrace();
            logger.error(e.getMessage());
        }
    }

**Comment by frankfliu** [comment, marked as answer]
@csxiang18 
You should be able to use: "jar:///dnn-model/model.zip", but the limitation is you need put your model.pt into a zip/tar file.

**Comment by haha00gou** [comment]
I have a similar problem , so do you have answer?

**Reply by frankfliu** [reply]
@haha00gou 
As I mentioned in previous answer, you can use `jar:///" as url to load the model from your jar file. Currently DJL only support load archive file from the jar, see examples:
https://github.com/awslabs/djl/blob/master/api/src/test/java/ai/djl/repository/JarRepositoryTest.java#L39

**Reply by haha00gou** [reply]
thank you
